static/app/views/performance/newTraceDetails/traceApi/useTrace.tsx

```diff
@@ -92,32 +101,105 @@ export function getTraceQueryParams(
   return queryParams;
 }
 
+function parseSampleSlug(
+  sample: string | undefined
+): {event_id: string; project_slug: string} | null {
+  if (!sample) {
+    return null;
+  }
+
+  const [project_slug, event_id] = sample.split(':');
+  return {project_slug, event_id};
+}
+
+function makeTraceFromTransaction(
+  event: EventTransaction | undefined
+): TraceSplitResults<TraceFullDetailed> | undefined {
+  if (!event) {
+    return undefined;
+  }
+
+  const traceContext = event.contexts?.trace;
+
+  const transaction = {
+    event_id: event.eventID,
+    generation: 0,
+    parent_event_id: '',
+    parent_span_id: traceContext?.parent_span_id ?? '',
+    performance_issues: [],
+    project_id: Number(event.projectID),
+    project_slug: event.projectSlug ?? '',
+    span_id: traceContext?.span_id ?? '',
+    timestamp: event.endTimestamp,
+    transaction: event.title,
+    'transaction.duration': (event.endTimestamp - event.startTimestamp) * 1000,
+    errors: [],
+    children: [],
+    start_timestamp: event.startTimestamp,
+    'transaction.op': traceContext?.op ?? '',
+    'transaction.status': traceContext?.status ?? '',
+    measurements: event.measurements ?? {},
+    tags: [],
+  };
+
+  return {transactions: [transaction], orphan_errors: []};
+}
+
 type UseTraceParams = {
   limit?: number;
 };
 
 const DEFAULT_OPTIONS = {};
 export function useTrace(
   options: Partial<UseTraceParams> = DEFAULT_OPTIONS
-): UseApiQueryResult<TraceSplitResults<TraceFullDetailed>, any> {
+): Pick<UseApiQueryResult<TraceSplitResults<TraceFullDetailed>, any>, 'data' | 'status'> {
   const filters = usePageFilters();
   const organization = useOrganization();
   const params = useParams<{traceSlug?: string}>();
-
   const queryParams = useMemo(() => {
     const query = qs.parse(location.search);
     return getTraceQueryParams(query, filters.selection, options);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [options]);
+  const sample = parseSampleSlug(queryParams.sampleSlug);
 
-  return useApiQuery(
+  // When projects don't have performance set up, we allow them to view a sample transaction.
+  // Now that a transaction always shows up as a part of the trace it is associate with, we
+  // we need to display it as a part of the trace view. Therefore, we fetch the sample transaction
+  // make a trace, with it as the only event in it.
+  const sampleQuery = useApiQuery<EventTransaction>(
```

Comment:
Since this is "demo" functionality, it's not great to have it pollute the code too much and I would prefer if we isolated it. 

A way to do that would be to create a useSampleTrace in this same file and then toggle the return from inside useTrace()